core/src/main/java/org/jivesoftware/sparkimpl/plugin/transcripts/HistoryTranscript.java

```diff
@@ -258,18 +266,47 @@ public final String buildString(List<HistoryMessage> messages){
 	private synchronized void display() {
 		try {
 			token.acquire();
-
 			if ((searchFilteredList.size() > 0) && (pageIndex <= searchFilteredList.size())) {
 				builder.append("<html><body><table cellpadding=0 cellspacing=0>");
 				builder.append(buildString(searchFilteredList.get(pageIndex-1).getMessages()));
-			}else{
+			} else {
 				// Handle no history
 				builder.replace(0, builder.length(), "");
 				builder.append("<b>")
 						.append(Res.getString("message.no.history.found"))
 						.append("</b>");
 			}
 			window.setText(builder.toString());
+
+			// highlighting searched text
+			try {
+				String text = searchQuery;
+				Document doc = window.getDocument();
+				String line;
+				if (text.length() > 0 && focusFlag) {
+					String str = doc.getText(1, doc.getLength());
+					BufferedReader buf = new BufferedReader(new StringReader(str));
+					int globalPos = 1;
+					while ((line = buf.readLine()) != null){
+						Matcher matcherTime = Pattern
+								.compile("\\(\\d\\d:\\d\\d:\\d\\d\\)\\s[a-zA-Z0-9]+:\\s")
+								.matcher(line);
+						if (matcherTime.find()) {
+							for (int i = matcherTime.end(); i + text.length() < line.length() + 1; i++) {
+								String match = line.substring(i, i + text.length());
+								if (text.toLowerCase().equals(match.toLowerCase())) {
```

Comment:
This does the same, but is somewhat easier on the eyes: ``text.equalsIgnoreCase( match )``